### ISSUE TITLE
Add Launch with Third Party to Tool Page

### ISSUE THREAD

**Issue by coverbeck** [ga4gh/dockstore#1705]
## Feature Request 

The Launch with buttons for FireCloud,  DNAstack, and DNAnexus only appear on the Workflow page.

If feasible, we should add those buttons to the Tool page as well.

Depending on the APIs those stacks are using to call back to Dockstore, this may or not be possible without any changes on the other stacks's side.

**Comment by denis-yuen** [comment]
Start with Terra

**Comment by coverbeck** [comment]
To add this for Terra, this will require a change on their side. Right now, for a workflow, we redirect to, for example:

`<terra base url>#import-tool/dockstore/github.com/DataBiosphere/topmed-workflows/UM_variant_caller_wdl:1.32.0`

The Terra page then makes an Ajax call to Dockstore, using staging as an example: `https://staging.dockstore.org:8443/api/ga4gh/v1/tools/%23workflow%2Fgithub.com%2FDataBiosphere%2Ftopmed-workflows%2FUM_variant_caller_wdl/versions/1.32.0/WDL/descriptor`. Note the prepending of `%23workflow%2F` that got inserted.

It is always inserted [here in the Terra code](https://github.com/DataBiosphere/terra-ui/blob/be2abe0f8ac2aac0ff99e6d783bfae9d7b50c7a8/src/libs/ajax.js#L98), which means we won't be able to specify a tool

**Comment by coverbeck** [comment]
Created a ticket for Terra, DataBiosphere/terra-ui#1435

**Comment by cricketsloan** [comment]
➤ Charles Overbeck commented:

D’oh\! Apparently they closed issues (they turned off the whole feature) on the repo, and the ticket from the previous comment either got dropped or moved to JIRA.

**Comment by coverbeck** [comment]
Logged a feature request for Terra [here](https://support.terra.bio/hc/en-us/community/posts/360047834291-Terra-Should-Support-Dockstore-Tools). They are now using this instead of GitHub issues.

**Comment by denis-yuen** [comment]
One thought, if the same mechanism is used for services ...

1) https://app.terra.bio/#import-tool/dockstore-tool/github.com/HumanCellAtlas/skylab/Optimus:terra-optimus implies https://app.terra.bio/#import-tool/dockstore-service/github.com/xenaorg/xena/Optimus:terra-optimus

2) https://app.terra.bio/#import-tool/dockstore/github.com/HumanCellAtlas/skylab/Optimus:terra-optimus?type=tool implies
https://app.terra.bio/#import-tool/dockstore/github.com/xenaorg/xena/Optimus:terra-optimus?type=service

The latter looks a bit cleaner

**Comment by denis-yuen** [comment]
https://support.terra.bio/hc/en-us/community/posts/360042846312-Import-json-file-automatically-when-creating-a-method-from-Dockstore would also be helpful to keep an eye on/help out if we can

**Comment by cricketsloan** [comment]
➤ Charles Overbeck commented:

No plans for immediate support by Terra: https://support.terra.bio/hc/en-us/community/posts/360047834291/comments/360006939931 ( https://support.terra.bio/hc/en-us/community/posts/360047834291/comments/360006939931 )

**Comment by denis-yuen** [comment]
Looks like yes

**Comment by coverbeck** [comment]
I believe this is doable today for at least SB-based platforms, since we pass them the full TRS url.

**Comment by coverbeck** [comment]
Put in as a 1.12 milestone as a stretch goal, but let's discuss in next backlog grooming. I think it might be really easy.

**Comment by coverbeck** [comment]
Tried a test with Ash, doesn't seem to work out of the box.

**Comment by melainalegaspi** [comment]
@coverbeck "blocked on response from Sevenbridges."
